addons/mass_mailing/models/mass_mailing.py

```diff
@@ -54,6 +54,50 @@ def _compute_contact_nbr(self):
         for mailing_list in self:
             mailing_list.contact_nbr = data.get(mailing_list.id, 0)
 
+    @api.multi
+    def copy(self, default=None):
+        recipients = self.env['mail.mass_mailing.contact'].search([('list_ids', 'in', self.ids), ('opt_out', '!=', True)])
+        mailing_list = super(MassMailingList, self).copy(default=default)
+        for recipient in recipients:
+            recipient.write({'list_ids': [(4, mailing_list.ids)]})
+        return mailing_list
+
+    def merge_massmail_list(self, mailing_lists):
```

Comment:
Do an api multi.

I don't understand the purpose of this method: make all recipients members of all mailing lists ? In my opinion the purpose of merging something is to have a single one. Actually you should merge contacts and lists here, and the wizard should call this method.

I think it should be

1/ find all contacts of all lists
2/ update contacts in destination mailing list
2/ archive other mailing lists (if not possible, add active field in order to be able to do so)